packages/ui-components/src/icons.tsx

```diff
@@ -68,4 +68,29 @@ export class IconUtil {
   static encode(icon: LabIcon): string {
     return 'data:image/svg+xml;utf8,' + encodeURIComponent(icon.svgstr);
   }
+
+  static colorize(
+    icon: LabIcon,
+    fillColor: string,
+    strokeColor?: string
+  ): LabIcon {
+    let svgstr = icon.svgstr;
+
+    if (fillColor) {
+      svgstr = svgstr.replace(/fill="[^(none)]+?"/gi, `fill="${fillColor}"`);
+    }
+    if (strokeColor) {
+      svgstr = svgstr.replace(
+        /stroke="[^(none)]+?"/gi,
+        `stroke="${strokeColor}"`
+      );
+    }
+
+    return LabIcon.resolve({
+      icon: {
+        name: `${icon.name}:${fillColor || ''}:${strokeColor || ''}`,
```

Comment:
we should tweak this so the `:` only are added if `fillColor` or `strokeColor` exist. Otherwise we could get a name with extra colons

Reply:
done